contracts/tokenomics/maker/src/contract.rs

```diff
@@ -452,12 +484,43 @@ fn distribute_astro(deps: DepsMut, env: Env, info: MessageInfo) -> Result<Respon
 /// # Params
 /// * **deps** is the object of type [`DepsMut`].
 ///
+/// * **env** is the object of type [`Env`].
+///
 /// * **cfg** is the object of type [`Config`].
 ///
 /// * **amount** is the object of type [`Uint128`].
-fn distribute(deps: Deps, cfg: &Config, amount: Uint128) -> Result<Vec<SubMsg>, ContractError> {
+fn distribute(
+    deps: DepsMut,
+    env: Env,
+    cfg: &mut Config,
+    mut amount: Uint128,
+) -> Result<Vec<SubMsg>, ContractError> {
     let mut result = vec![];
 
+    if !cfg.rewards_enabled {
+        cfg.pre_upgrade_astro_amount = amount;
+        CONFIG.save(deps.storage, cfg)?;
+        return Ok(result);
+    } else if !cfg.pre_upgrade_astro_amount.is_zero() {
+        let astro_distribution_portion = cfg
+            .astro_distribution_portion
+            .unwrap_or(cfg.pre_upgrade_astro_amount / (Uint128::from(cfg.pre_upgrade_blocks)));
+        cfg.astro_distribution_portion = Some(astro_distribution_portion);
```

Comment:
Seems like we don't need to store it to config, it's fine to calculate on the fly